scripts/populate_ingest/populate_ingest_from_nxn.py

```diff
@@ -0,0 +1,178 @@
+
+    def __compare_on_accession__(self):
+        ingest_data_accessions = set(itertools.chain.from_iterable(
+            [data.get("insdc_project_accessions") for data in self.ingest_data if
+             data.get("insdc_project_accessions")] +
+            [data.get("geo_series_accessions") for data in self.ingest_data if data.get("geo_series_accessions")] +
+            [data.get("ega_accessions") for data in self.ingest_data if data.get("ega_accessions")] +
+            [data.get("dbgap_accessions") for data in self.ingest_data if data.get("dbgap_accessions")] +
+            [data.get("array_express_accessions") for data in self.ingest_data if
+             data.get("array_express_accessions")] +
+            [data.get("biostudies_accessions") for data in self.ingest_data if data.get("biostudies_accessions")] +
+            [data.get("insdc_study_accessions") for data in self.ingest_data if data.get("insdc_study_accessions")]))
+
+        filter_accessions = self.nxn_data.get_values('Data location') - ingest_data_accessions
+        self.nxn_data.data = [row for row in self.nxn_data.data if self.nxn_data.get_value(row, 'Data location') in filter_accessions
+                              or not self.nxn_data.get_value(row, 'Data location')]
+
+    def __compare_on_title__(self):
+        ingest_data_pubs = list(itertools.chain.from_iterable([data.get("publications") for data in self.ingest_data if data.get("publications")]))
+        ingest_data_titles = set([reformat_title(pub.get("title")) for pub in ingest_data_pubs if pub.get("title")])
+
+        filter_titles = {reformat_title(title) for title in self.nxn_data.get_values('Title')} - ingest_data_titles
+        filter_titles = {title for title in filter_titles if not any([get_distance_metric(title, tracking_title)
+                                                                >= 97 for tracking_title in ingest_data_titles])}
+        self.nxn_data.data = [row for row in self.nxn_data.data if reformat_title(self.nxn_data.get_value(row, 'Title')) in filter_titles]
+
+    def compare(self):
+        self.__compare_on_doi__()
+        self.__compare_on_accession__()
+        self.__compare_on_title__()
+
+    def filter(self):
+        self.nxn_data.data = [row for row in self.nxn_data.data if
+                         self.nxn_data.get_value(row, 'Organism').lower() in ["human", "human, mouse", "mouse, human"]]
+        self.nxn_data.data = [row for row in self.nxn_data.data if
+                         any([tech.strip() in ["chromium", "drop-seq", "dronc-seq", "smart-seq2", "smarter",
+                                               "smarter (C1)"] for tech in
+                              self.nxn_data.get_value(row, 'Technique').lower().split("&")])]
+        self.nxn_data.data = [row for row in self.nxn_data.data if self.nxn_data.get_value(row, 'Measurement').lower() == 'rna-seq']
+
+    def __convert__(self, nxn_data_row) -> dict:
+        ingest_project = {
```

Comment:
This could be refactored to be made of functions in appropriate level of abstraction:
- make_ingest_Project
- add_publication_info
- add_accesstion

and so on.